wandb/sdk/wandb_run.py

```diff
@@ -756,6 +757,10 @@ def restore(
         files = api_run.files([name])
         if len(files) == 0:
             return None
+        # if the file does not exist, the file has an md5 of 0
+        if files[0].md5 == "0":
```

Comment:
this seems like a nasty side effect of our graphql interface.  is there something better?:
```
FILE_FRAGMENT = '''fragment RunFilesFragment on Run {
    files(names: $fileNames, after: $fileCursor, first: $fileLimit) {
        edges {
            node {
                id
                name
                url(upload: $upload)
                sizeBytes
                mimetype
                updatedAt
                md5
            }
```

Reply:
I wasn't sure if that was the desired functionality, since the resolver has a specific flow for returning a file that is not found. In this flow, the returned file has the size set to 0 and the updatedAt set to `None`.
[Source](https://github.com/wandb/core/blob/master/services/gorilla/resolver/run.go#L559-L577):
```go
for i, n := range flatNames {
	f, ok := fmap[n]
	if ok {
		frs[i] = &FileResolver{
			baseResolver: r.baseResolver,
			file:         f,
		}
	} else {
		frs[i] = &NamedFileResolver{
			baseResolver:       r.baseResolver,
			projectStorageName: p.StorageKey,
			entityName:         e.Name,
			projectName:        p.Name,
			runName:            r.run.Name,
			fileName:           n,
			projectID:          r.run.ProjectID,
		}
	}
}
```

Reply:
It also has md5 set to "0"
```go
func (f *NamedFileResolver) MD5() *string {
	zero := "0"
	return &zero
}
```